File: SuboptimalProcedureLearner/SuboptimalProcedureLearner/SuboptimalProcedureLearner/Operators/OperatorConstant.cs
using System;
using System.Collections.Generic;

using Datastructures;

namespace SuboptimalProcedureLearner.SuboptimalProcedureLearner.Operators
{
    class OperatorConstant : Operator
    {
        override public IntrospectWiringInfo introspectWiring(OperatorInstance instance, int[] inputIndices)
        {
            throw new NotImplementedException();
        }

        public override OperatorInstance createOperatorInstance()
        {
            return new OperatorInstance(this);
        }
        
        override public void setParameterOperatorInstances(OperatorInstance instance, List<OperatorInstance> parameterInstances)
        {
            System.Diagnostics.Debug.Assert(parameterInstances.Count == 1);

            instance.constants = new Variadic[1];
            instance.constants[0] = parameterInstances[0].constants[0];
        }
        
        override public void initializeOperatorInstance(OperatorInstance instance)
        {
        }


        override public ExecutionResult executeSingleStep(OperatorInstance instance)
        {
            return ExecutionResult.createResultInstance(instance.constants[0]);
        }

        // gets called from the executive if a (requested execution) operator(instance) was executed
        override public void feedOperatorInstanceResult(OperatorInstance instance, Variadic result)
        {
            int x = 0;
        }




        override public void operationCleanup(OperatorInstance instance)
        {
        }

        override public bool isScaffold()
        {
            return false;
        }

        override public string getShortName()
        {
            return "OperatorConstant";
        }
    }
}

File: src/java/ptrman/agi-x0/src/java/SuboptimalProcedureLearner/Operators/OperatorConstant.java
package SuboptimalProcedureLearner.Operators;

import Datastructures.Variadic;
import SuboptimalProcedureLearner.Operator;
import SuboptimalProcedureLearner.OperatorInstance;
import ptrman.misc.Assert;
import sun.reflect.generics.reflectiveObjects.NotImplementedException;

import java.util.List;

public class OperatorConstant  extends Operator {
    public OperatorConstant() {
        super();
    }

    public IntrospectWiringInfo introspectWiring(OperatorInstance instance, int[] inputIndices) {
        throw new NotImplementedException();
    }

    public OperatorInstance createOperatorInstance() {
        return new OperatorInstance(this);
    }

    public void setParameterOperatorInstances(OperatorInstance instance, List<OperatorInstance> parameterInstances) {
        Assert.Assert(parameterInstances.size() == 1, "");
        instance.constants = new Variadic[1];
        instance.constants[0] = parameterInstances.get(0).constants[0];
    }

    public void initializeOperatorInstance(OperatorInstance instance) {
    }

    public ExecutionResult executeSingleStep(OperatorInstance instance) {
        return ExecutionResult.createResultInstance(instance.constants[0]);
    }

    // gets called from the executive if a (requested execution) operator(instance) was executed
    public void feedOperatorInstanceResult(OperatorInstance instance, Variadic result) {
        int x = 0;
    }

    public void operationCleanup(OperatorInstance instance) {
    }

    public String getShortName() {
        return this.getClass().getName();
    }
}
